agent/agentcontainers/containers_dockercli.go

```diff
@@ -279,63 +284,58 @@ func (dcl *DockerCLILister) List(ctx context.Context) (codersdk.WorkspaceAgentLi
 		return codersdk.WorkspaceAgentListContainersResponse{}, xerrors.Errorf("run docker inspect: %w", err)
 	}
 
-	for _, in := range ins {
-		out, warns := convertDockerInspect(in)
-		res.Warnings = append(res.Warnings, warns...)
-		res.Containers = append(res.Containers, out)
-	}
-
-	if dockerPsStderr != "" {
-		res.Warnings = append(res.Warnings, dockerPsStderr)
-	}
 	if dockerInspectStderr != "" {
 		res.Warnings = append(res.Warnings, dockerInspectStderr)
 	}
 
+	outs, warns, err := convertDockerInspect(ins)
+	if err != nil {
+		return codersdk.WorkspaceAgentListContainersResponse{}, xerrors.Errorf("convert docker inspect output: %w", err)
+	}
+	res.Warnings = append(res.Warnings, warns...)
+	res.Containers = append(res.Containers, outs...)
+
 	return res, nil
 }
 
 // runDockerInspect is a helper function that runs `docker inspect` on the given
 // container IDs and returns the parsed output.
 // The stderr output is also returned for logging purposes.
-func runDockerInspect(ctx context.Context, execer agentexec.Execer, ids ...string) ([]dockerInspect, string, error) {
+func runDockerInspect(ctx context.Context, execer agentexec.Execer, ids ...string) (stdout, stderr string, err error) {
 	var stdoutBuf, stderrBuf bytes.Buffer
 	cmd := execer.CommandContext(ctx, "docker", append([]string{"inspect"}, ids...)...)
 	cmd.Stdout = &stdoutBuf
 	cmd.Stderr = &stderrBuf
-	err := cmd.Run()
-	stderr := strings.TrimSpace(stderrBuf.String())
+	err = cmd.Run()
+	stdout = strings.TrimSpace(stdoutBuf.String())
+	stderr = strings.TrimSpace(stderrBuf.String())
 	if err != nil {
-		return nil, stderr, err
-	}
-
-	var ins []dockerInspect
-	if err := json.NewDecoder(&stdoutBuf).Decode(&ins); err != nil {
-		return nil, stderr, xerrors.Errorf("decode docker inspect output: %w", err)
+		return stdout, stderr, err
 	}
 
-	return ins, stderr, nil
+	return stdoutBuf.String(), stderr, nil
```

Comment:
Is it an oversight that stdoutBuf is accessed here instead of stdout? If not, an explanation would be good.

Reply:
Likely an oversight!